public/components/ui_action_context_menu.tsx

```diff
@@ -39,6 +42,35 @@ export const ActionContextMenu = (props: Props) => {
   });
   const resultSummaryEnabled = useObservable(props.resultSummaryEnabled$, false);
   const isQuerySummaryCollapsed = useObservable(props.isQuerySummaryCollapsed$, false);
+  const isSummaryAgentAvailable = useObservable(props.isSummaryAgentAvailable$, false);
+
+  async function checkAgentsExist(
+    http: HttpSetup,
+    agentConfigName: string | string[],
+    dataSourceId?: string
+  ) {
+    const response = await http.get(AGENT_API.CONFIG_EXISTS, {
+      query: { agentConfigName, dataSourceId },
+    });
+    return response;
+  }
+
+  useEffect(() => {
+    const fetchSummaryAgent = async () => {
+      try {
+        const summaryAgentStatus = await checkAgentsExist(
```

Comment:
Do we need to `props.isSummaryAgentAvailable$.next(false)` before making the call to check if agent exists? if we switch from data source A(agent configured) to data source B (agent not configured) and the call somehow failed, the status will be incorrect.